clang/lib/Sema/SemaDeclCXX.cpp

```diff
@@ -9767,7 +9767,9 @@ bool Sema::ShouldDeleteSpecialMember(CXXMethodDecl *MD,
     return false;
   CXXRecordDecl *RD = MD->getParent();
   assert(!RD->isDependentType() && "do deletion after instantiation");
-  if (!LangOpts.CPlusPlus || (!LangOpts.CPlusPlus11 && !RD->isLambda()) ||
+  if (!LangOpts.CPlusPlus ||
```

Comment:
are there cases where `!LangOpts.CPlusPlus` is true here?

Reply:
I wouldn't think so since we have a CXXMethodDecl and CXXRecordDecl. It was added with #73376 which changed it from`!CPlusPlus11` when adding C++03 lambdas

Reply:
I would try to remove it and see if tests pass (which i suspect they will)